### ISSUE TITLE
mgr daemons created by ceph-ansible are not visible to ceph

### ISSUE THREAD

**Issue by micmoyles** [ceph/ceph-ansible#5321]
[ansible-github.log](https://github.com/ceph/ceph-ansible/files/4529673/ansible-github.log)

**Bug Report**

I have 3 mgr daemons running but ceph is not aware of them.
```
[root@bvcephtest05 ~]# ceph status
  cluster:
    id:     81ae6812-607f-4205-9484-2d063a9c4431
    health: HEALTH_WARN
            clock skew detected on mon.bvcephtest04, mon.bvcephtest05

  services:
    mon: 3 daemons, quorum bvcephtest03,bvcephtest04,bvcephtest05 (age 5m)
    mgr: no daemons active
    osd: 0 osds: 0 up, 0 in

  data:
    pools:   0 pools, 0 pgs
    objects: 0 objects, 0 B
    usage:   0 B used, 0 B / 0 B avail
    pgs:


[root@bvcephtest05 ~]# systemctl status ceph-mgr
ceph-mgr@bvcephtest05.service  ceph-mgr.target
[root@bvcephtest05 ~]# systemctl status ceph-mgr@bvcephtest05.service
● ceph-mgr@bvcephtest05.service - Ceph cluster manager daemon
   Loaded: loaded (/usr/lib/systemd/system/ceph-mgr@.service; enabled; vendor preset: disabled)
   Active: active (running) since Fri 2020-04-24 11:34:35 BST; 5min ago
 Main PID: 50595 (ceph-mgr)
    Tasks: 21 (limit: 23597)
   Memory: 117.7M
   CGroup: /system.slice/system-ceph\x2dmgr.slice/ceph-mgr@bvcephtest05.service
           └─50595 /usr/bin/ceph-mgr -f --cluster ceph --id bvcephtest05 --setuser ceph --setgroup ceph

```
Apr 24 11:34:35 bvcephtest05 systemd[1]: Started Ceph cluster manager daemon..`

What you expected to happen:

The mgr daemons to be detected :) 

How to reproduce it (minimal and precise):

This is how I have defined my inventory. My hosts are all Centos8 VMs.
```
cat  inventory
[mons]
bvcephtest03
bvcephtest04
bvcephtest05
[mgrs]
bvcephtest03
bvcephtest04
bvcephtest05

[osds]
bvcephtest01
bvcephtest03
bvcephtest04
bvcephtest05

[rgws]
bvcephtest01

[grafana-server]
bvcephtest01
```

I have tried this a number of times and get the same result. I've used new VMs and run the purge-cluster playbook to start the entire process from scratch. 

The run fails at the same task every time
`TASK [ceph-mgr : wait for all mgr to be up]`
Share your group_vars files, inventory and **full** ceph-ansibe log

**Environment**:
* OS (e.g. from /etc/os-release):
```
NAME="CentOS Linux"
VERSION="8 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="CentOS Linux 8 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:8"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-8"
CENTOS_MANTISBT_PROJECT_VERSION="8"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="8"
```
* Kernel (e.g. `uname -a`):
`Linux bvcephtest05 4.18.0-147.8.1.el8_1.x86_64 #1 SMP Thu Apr 9 13:49:54 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`
* Docker version if applicable (e.g. `docker version`):
NA
* Ansible version (e.g. `ansible-playbook --version`):
ansible-playbook 2.9.6
* ceph-ansible version (e.g. `git head or tag or stable branch`):
stable-5.0
* Ceph version (e.g. `ceph -v`):
ceph version 15.2.1 (9fd2f65f91d9246fae2c841a6222d34d121680ee) octopus (stable)

*Group vars*
grep -vE '^#' group_vars/all.yml | uniq
```yaml
---
centos_package_dependencies:
  - epel-release
  - python3-libselinux
ntp_service_enabled: true
ceph_origin: repository
ceph_repository: community
ceph_stable_release: octopus
ceph_stable_redhat_distro: el8
monitor_address_block: 10.64.6.0/24
public_network: 10.64.6.0/24
cluster_network: "{{ public_network | regex_replace(' ', '') }}"
radosgw_address: 10.64.6.11
dashboard_enabled: True
dashboard_protocol: http
dashboard_port: 8443
dashboard_admin_user: admin
dashboard_admin_user_ro: false
dashboard_admin_password: p@ssw0rd
grafana_admin_user: admin
grafana_admin_password: admin
```
grep -vE '^#' group_vars/osds.yml | uniq
```yaml
---
devices:
  - /dev/sdb
```

**Comment by dsavineau** [comment]
Share your group_vars files, inventory and full ceph-ansibe log

**Comment by micmoyles** [comment]
> Share your group_vars files, inventory and full ceph-ansibe log

They are all included now. I haven't attached the full contents of group vars as they are most commented out, if the whole files are better please let me know.

**Comment by dsavineau** [comment]
> > Share your group_vars files, inventory and full ceph-ansibe log
> 
> They are all included now. I haven't attached the full contents of group vars as they are most commented out, if the whole files are better please let me know.

egrep -v '^(#|$|dummy)' group_vars/*.yml ;)

**Comment by micmoyles** [comment]
> > > Share your group_vars files, inventory and full ceph-ansibe log
> > 
> > 
> > They are all included now. I haven't attached the full contents of group vars as they are most commented out, if the whole files are better please let me know.
> 
> `egrep -v '^(#|$|dummy)' group_vars/*.yml` ;)

```
  egrep -v '^(#|$|dummy)' group_vars/*.yml
group_vars/all.yml:---
group_vars/all.yml:centos_package_dependencies:
group_vars/all.yml:  - epel-release
group_vars/all.yml:  - python3-libselinux
group_vars/all.yml:ntp_service_enabled: true
group_vars/all.yml:ceph_origin: repository
group_vars/all.yml:ceph_repository: community
group_vars/all.yml:ceph_stable_release: octopus
group_vars/all.yml:ceph_stable_redhat_distro: el8
group_vars/all.yml:monitor_address_block: 10.64.6.0/24
group_vars/all.yml:public_network: 10.64.6.0/24
group_vars/all.yml:cluster_network: "{{ public_network | regex_replace(' ', '') }}"
group_vars/all.yml:radosgw_address: 10.64.6.11
group_vars/all.yml:dashboard_enabled: True
group_vars/all.yml:dashboard_protocol: http
group_vars/all.yml:dashboard_port: 8443
group_vars/all.yml:dashboard_admin_user: admin
group_vars/all.yml:dashboard_admin_user_ro: false
group_vars/all.yml:dashboard_admin_password: p@ssw0rd
group_vars/all.yml:grafana_admin_user: admin
group_vars/all.yml:grafana_admin_password: admin
group_vars/mons.yml:---
group_vars/osds.yml:---
group_vars/osds.yml:devices:
group_vars/osds.yml:  - /dev/sdb
group_vars/rgws.yml:---
```

**Comment by dsavineau** [comment]
Would it be possible to get the ceph-mgr journald log from one of the mgr node ?

From the ceph-ansible execution perspective everything seems fine.

You mentioned that you're using the stable-5.0 branch. Could you tell me the exact sha1 commit ?

**Comment by micmoyles** [comment]
Sure 

```
 journalctl -u  ceph-mgr@bvcephtest05.service
-- Logs begin at Fri 2020-04-24 12:26:42 BST, end at Fri 2020-04-24 16:20:30 BST. --
Apr 24 12:26:50 bvcephtest05 systemd[1]: Started Ceph cluster manager daemon.
```

```
cat ceph-mgr.bvcephtest05.log
2020-04-24T11:34:35.567+0100 7f2538c85040  0 set uid:gid to 167:167 (ceph:ceph)
2020-04-24T11:34:35.567+0100 7f2538c85040  0 ceph version 15.2.1 (9fd2f65f91d9246fae2c841a6222d34d121680ee) octopus (stable), process ceph-mgr, pid 50595
2020-04-24T11:34:35.567+0100 7f2538c85040  0 pidfile_write: ignore empty --pid-file
2020-04-24T11:34:35.590+0100 7f2538c85040  1 mgr[py] Loading python module 'alerts'
2020-04-24T11:34:35.672+0100 7f2538c85040  1 mgr[py] Loading python module 'balancer'
2020-04-24T11:34:35.728+0100 7f2538c85040  1 mgr[py] Loading python module 'crash'
2020-04-24T11:34:35.785+0100 7f2538c85040  1 mgr[py] Loading python module 'dashboard'
2020-04-24T11:34:36.385+0100 7f2538c85040  1 mgr[py] Loading python module 'devicehealth'
2020-04-24T11:34:36.434+0100 7f2538c85040  1 mgr[py] Loading python module 'diskprediction_local'
2020-04-24T12:18:57.575+0100 7f3afe86e040  0 set uid:gid to 167:167 (ceph:ceph)
2020-04-24T12:18:57.575+0100 7f3afe86e040  0 ceph version 15.2.1 (9fd2f65f91d9246fae2c841a6222d34d121680ee) octopus (stable), process ceph-mgr, pid 53708
2020-04-24T12:18:57.575+0100 7f3afe86e040  0 pidfile_write: ignore empty --pid-file
2020-04-24T12:18:57.607+0100 7f3afe86e040  1 mgr[py] Loading python module 'alerts'
2020-04-24T12:18:57.719+0100 7f3afe86e040  1 mgr[py] Loading python module 'balancer'
2020-04-24T12:18:57.803+0100 7f3afe86e040  1 mgr[py] Loading python module 'crash'
2020-04-24T12:18:57.888+0100 7f3afe86e040  1 mgr[py] Loading python module 'dashboard'
2020-04-24T12:18:58.602+0100 7f3afe86e040  1 mgr[py] Loading python module 'devicehealth'
2020-04-24T12:18:58.676+0100 7f3afe86e040  1 mgr[py] Loading python module 'diskprediction_local'
2020-04-24T12:26:58.180+0100 7fcac6a5a040  0 set uid:gid to 167:167 (ceph:ceph)
2020-04-24T12:26:58.180+0100 7fcac6a5a040  0 ceph version 15.2.1 (9fd2f65f91d9246fae2c841a6222d34d121680ee) octopus (stable), process ceph-mgr, pid 1753
2020-04-24T12:26:58.180+0100 7fcac6a5a040  0 pidfile_write: ignore empty --pid-file
2020-04-24T12:26:58.235+0100 7fcac6a5a040  1 mgr[py] Loading python module 'alerts'
2020-04-24T12:26:58.495+0100 7fcac6a5a040  1 mgr[py] Loading python module 'balancer'
2020-04-24T12:26:58.609+0100 7fcac6a5a040  1 mgr[py] Loading python module 'crash'
2020-04-24T12:26:58.781+0100 7fcac6a5a040  1 mgr[py] Loading python module 'dashboard'
2020-04-24T12:26:59.759+0100 7fcac6a5a040  1 mgr[py] Loading python module 'devicehealth'
2020-04-24T12:26:59.809+0100 7fcac6a5a040  1 mgr[py] Loading python module 'diskprediction_local'
```

HEAD of my repo
commit 47d7819bbecf90cea11ab16c619b0ec8220f6c6e (HEAD -> stable-5.0, origin/stable-5.0)

From the ceph-ansible perspective everything runs without error until the step where is tries to verify the mgrs are active. In a previous task to start the mgr daemons everything runs smoothly. 

When I dump the state of the mgr I notice that the active_addrs is empty and that the range is unrecognised.

```
[root@bvcephtest05 ceph-bvcephtest05]# ceph --cluster ceph mgr dump
{
    "epoch": 1,
    "active_gid": 0,
    "active_name": "",
    "active_addrs": {
        "addrvec": []
    },
    "active_addr": "(unrecognized address family 0)/0",
    "active_change": "0.000000",
    "active_mgr_features": 0,
    "available": false,
    "standbys": [],
    "modules": [
        "iostat",
        "restful"
    ],
    "available_modules": [],
    "services": {},
    "always_on_modules": {
        "nautilus": [
            "balancer",
            "crash",
            "devicehealth",
            "orchestrator_cli",
            "progress",
            "rbd_support",
            "status",
            "volumes"
        ],
        "octopus": [
            "balancer",
            "crash",
            "devicehealth",
            "orchestrator",
            "osd_support",
            "pg_autoscaler",
            "progress",
            "rbd_support",
            "status",
            "telemetry",
            "volumes"
        ],
        "last_failure_osd_epoch": 0,
        "active_clients": []
    }
}
```

**Comment by micmoyles** [comment]
Thanks for linking the other issue. 
Here is some more info from my side
*ceph status*
```
[root@bvcephtest05 ceph]# ceph -s
  cluster:
    id:     81ae6812-607f-4205-9484-2d063a9c4431
    health: HEALTH_WARN
            clock skew detected on mon.bvcephtest04, mon.bvcephtest05

  services:
    mon: 3 daemons, quorum bvcephtest03,bvcephtest04,bvcephtest05 (age 4h)
    mgr: no daemons active
    osd: 0 osds: 0 up, 0 in

  data:
    pools:   0 pools, 0 pgs
    objects: 0 objects, 0 B
    usage:   0 B used, 0 B / 0 B avail
    pgs:
```
*mgr daemon config*
```
 ceph daemon  mgr.bvcephtest05 config  show | grep 10.64
    "cluster_network": "10.64.6.0/24",
    "mon_host": "[v2:10.64.6.8:3300,v1:10.64.6.8:6789],[v2:10.64.6.9:3300,v1:10.64.6.9:6789],[v2:10.64.6.10:3300,v1:10.64.6.10:6789]",
    "public_addr": "v2:10.64.6.10:0/0",
    "public_network": "10.64.6.0/24",
```

Is there anything else that would be helpful?

**Comment by dsavineau** [comment]
The only thing I'm thinking about could the firewalld rules so if you could check this on all nodes.
This should be correctly configured by ceph-ansible but just in case...

**Comment by micmoyles** [comment]
> The only thing I'm thinking about could the firewalld rules so if you could check this on all nodes.
> This should be correctly configured by ceph-ansible but just in case...

There are no firewall or iptables rules in place. I can reach every socket from each node

```
 telnet bvcephtest03 3300
Trying 10.64.6.8...
Connected to bvcephtest03.
Escape character is '^]'.
ceph v2
                 ^]
telnet> q
Connection closed.
```

Should I be seeing the mgr listening for incoming connections? Right now they are all only establishing connections to their local mon.
```
[root@bvcephtest05 ceph]# netstat -tunapl | grep ceph-mgr
tcp        0      0 10.64.6.10:36468        10.64.6.10:3300         ESTABLISHED 1753/ceph-mgr
```

**Comment by dsavineau** [comment]
> There are no firewall or iptables rules in place. I can reach every socket from each node

Can you double checked this ?
According to the ansible logs, firewalld is started and rules are added.
By default, ceph-ansible will try to start and configure firewalld unless you set `configure_firewall: false ` in the configurarion

https://github.com/ceph/ceph-ansible/blob/stable-5.0/roles/ceph-infra/tasks/configure_firewall.yml

> Should I be seeing the mgr listening for incoming connections? Right now they are all only establishing connections to their local mon.

ceph-mgr daemons aren't binding on a tcp socket unless you have some modules enabled (like dashboard/prometheus).
But you should see connections to other mons

```console
# netstat -lntupa|grep ceph-mgr
tcp        0      0 192.168.100.8:8443      0.0.0.0:*               LISTEN      4365/ceph-mgr       
tcp        0      0 192.168.100.8:45788     192.168.100.15:6800     ESTABLISHED 4365/ceph-mgr       
tcp        0      0 192.168.100.8:60780     192.168.100.8:3300      ESTABLISHED 4365/ceph-mgr       
tcp6       0      0 :::9283                 :::*                    LISTEN      4365/ceph-mgr
```

**Comment by dsavineau** [comment]
> ceph-mgr daemons aren't binding on a tcp socket unless you have some modules enabled (like dashboard/prometheus).

small correction : only the active mgr daemon is binding on a tcp socket (6800/6801).

```console
$ sudo netstat -lntupa|grep ceph-mgr
tcp        0      0 192.168.100.15:6800     0.0.0.0:*               LISTEN      28626/ceph-mgr      
tcp        0      0 192.168.100.15:6801     0.0.0.0:*               LISTEN      28626/ceph-mgr      
tcp        0      0 192.168.100.15:8443     0.0.0.0:*               LISTEN      28626/ceph-mgr      
tcp        0      0 192.168.100.15:40152    192.168.100.15:6800     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:37716    192.168.100.15:3300     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.15:40160    ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.8:45834     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:37704    192.168.100.15:3300     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.13:47652    ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.15:40154    ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.15:40102    ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.8:45790     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.15:40110    ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:38926    192.168.100.15:3300     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.13:47674    ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:43236    192.168.100.8:6800      ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.15:40158    ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.13:47670    ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:48680    192.168.100.13:6800     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.8:45836     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.15:40162    ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.8:45832     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.13:47656    ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.15:40150    ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.15:40152    ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:40102    192.168.100.15:6800     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:48892    192.168.100.15:6802     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.13:47676    ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.15:40156    ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:37690    192.168.100.15:3300     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.15:40134    ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:48678    192.168.100.13:6800     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:40150    192.168.100.15:6800     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.8:45786     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:57688    192.168.100.15:6810     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:40134    192.168.100.15:6800     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:6800     192.168.100.13:47672    ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:60268    192.168.100.13:6808     ESTABLISHED 28626/ceph-mgr      
tcp        0      0 192.168.100.15:39770    192.168.100.8:6808      ESTABLISHED 28626/ceph-mgr      
tcp6       0      0 :::9283                 :::*                    LISTEN      28626/ceph-mgr
```

**Comment by micmoyles** [comment]
Sorry about that, you are correct firewalld was configured. I have disabled it and restarted all daemons but still see the same issue. I do not have my own firewall in place on these machines.

Interestingly, none of my mgrs are listening on 6800/6801 or any other port for that matter.

```
[root@bvcephtest03 ~]# netstat -tunapl | grep ceph-mgr
tcp        0      0 10.64.6.8:59174         10.64.6.10:3300         ESTABLISHED 15117/ceph-mgr
[root@bvcephtest04 ~]# netstat -tunapl | grep ceph-mgr
tcp        0      0 10.64.6.9:46566         10.64.6.10:3300         ESTABLISHED 15041/ceph-mgr
[root@bvcephtest05 ~]# netstat -tunapl | grep ceph-mgr
tcp        0      0 10.64.6.10:36656        10.64.6.10:3300         ESTABLISHED 15851/ceph-mgr
```

**Comment by micmoyles** [comment]
My mgr daemons to not respond to ceph tell commands 
```
[root@bvcephtest04 ~]# ceph tell mgr help

 Tell mgr commands
 =================
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/ceph_argparse.py", line 1391, in send_command
    cluster.mgr_command, cmd, inbuf, timeout=timeout, target=name)
  File "/usr/lib/python3.6/site-packages/ceph_argparse.py", line 1339, in run_in_thread
    raise Exception("timed out")
Exception: timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/ceph_argparse.py", line 1492, in json_command
    inbuf, timeout, verbose)
  File "/usr/lib/python3.6/site-packages/ceph_argparse.py", line 1349, in send_command_retry
    return send_command(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/ceph_argparse.py", line 1448, in send_command
    raise RuntimeError('"{0}": exception {1}'.format(cmd, e))
RuntimeError: "['{"prefix": "get_command_descriptions"}']": exception timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/ceph", line 1275, in <module>
    retval = main()
  File "/usr/bin/ceph", line 1060, in main
    return do_extended_help(parser, childargs, target, None)
  File "/usr/bin/ceph", line 414, in do_extended_help
    return help_for_target(target, partial)
  File "/usr/bin/ceph", line 398, in help_for_target
    timeout=10)
  File "/usr/lib/python3.6/site-packages/ceph_argparse.py", line 1496, in json_command
    raise RuntimeError('"{0}": exception {1}'.format(argdict, e))
RuntimeError: "None": exception "['{"prefix": "get_command_descriptions"}']": exception timed out 
```

**Comment by dsavineau** [comment]
> Exception: timed out

So it looks like a network issue.

Would it be possible to run a tcpdump in parallel and see the flows ?

**Comment by dsavineau** [comment]
I tried to replicate your setup with the same configuration (except the network cidr) and firewalld enable but everything went well

```console
TASK [show ceph status for cluster ceph] *****************************************************************************************************************************************************************************************************
task path: /home/ds/Workspace/mgrwait/site.yml.sample:457
Friday 24 April 2020  14:55:52 -0400 (0:00:01.946)       0:21:38.053 ********** 
ok: [bvcephtest03 -> 10.41.11.143] => 
  msg:
  - '  cluster:'
  - '    id:     2c427774-ecef-435f-bbea-48a9ef538bb4'
  - '    health: HEALTH_OK'
  - ' '
  - '  services:'
  - '    mon: 3 daemons, quorum bvcephtest05,bvcephtest03,bvcephtest04 (age 15m)'
  - '    mgr: bvcephtest05(active, starting, since 2s), standbys: bvcephtest04, bvcephtest03'
  - '    osd: 4 osds: 4 up (since 9m), 4 in (since 9m)'
  - '    rgw: 1 daemon active (bvcephtest01.rgw0)'
  - ' '
  - '  task status:'
  - ' '
  - '  data:'
  - '    pools:   5 pools, 105 pgs'
  - '    objects: 189 objects, 5.0 KiB'
  - '    usage:   4.1 GiB used, 196 GiB / 200 GiB avail'
  - '    pgs:     105 active+clean'
  - ' '
META: ran handlers
META: ran handlers

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
bvcephtest01               : ok=278  changed=60   unreachable=0    failed=0    skipped=350  rescued=0    ignored=0   
bvcephtest03               : ok=351  changed=49   unreachable=0    failed=0    skipped=446  rescued=0    ignored=0   
bvcephtest04               : ok=287  changed=43   unreachable=0    failed=0    skipped=407  rescued=0    ignored=0   
bvcephtest05               : ok=297  changed=45   unreachable=0    failed=0    skipped=404  rescued=0    ignored=0   


INSTALLER STATUS *****************************************************************************************************************************************************************************************************************************
Install Ceph Monitor           : Complete (0:01:52)
Install Ceph Manager           : Complete (0:03:18)
Install Ceph OSD               : Complete (0:02:43)
Install Ceph RGW               : Complete (0:01:17)
Install Ceph Dashboard         : Complete (0:02:11)
Install Ceph Grafana           : Complete (0:03:18)
Install Ceph Node Exporter     : Complete (0:02:00)
```

**Comment by micmoyles** [comment]
@dsavineau, thanks for replicating. Over the weekend I altered the config a few times and managed to get something working. It seems my issue was that 3 mgr daemons are somehow unable to elect active and passive between them. My hunch is that this is caused by some old configs that were left on the machine between my installations and purges (the machines were brand new VMs before this). I added a clean machine to the cluster and used it as the only mon/mgr which allowed the cluster to come up healthy. 

If I get some time I'll try re-installing a mon and mgr back onto the troublesome machines in an effort to understand what the cause of this was. I'll close this issue for now and put it down to user error. Thanks for the assistance.
